templates/skeleton/app/components/Header.tsx

```diff
@@ -159,15 +163,19 @@ function CartToggle({cart}: Pick<HeaderProps, 'cart'>) {
   return (
     <Suspense fallback={<CartBadge count={null} />}>
       <Await resolve={cart}>
-        {(cart) => {
-          if (!cart) return <CartBadge count={0} />;
-          return <CartBadge count={cart.totalQuantity || 0} />;
-        }}
+        <CartBanner />
       </Await>
     </Suspense>
   );
 }
 
+function CartBanner() {
+  const cart = useAsyncValue();
+  const optimisticCart = useOptimisticCart(cart);
+  const linesCount = optimisticCart?.lines?.nodes?.length || 0;
```

Comment:
This only counts the lines, instead of summing line quantities. `useOptimisticCart` was updated recently to have a `totalQuantity` property - you could lean on that here.

Reply:
I can confirm that it's fixed with the new release!